### ISSUE TITLE
Investigate cleaner FIDO2 event-loop management

### ISSUE THREAD

**Issue by matejcik** [trezor/trezor-firmware#729]
The FIDO2 module needs to do some wild things with low-level synchronization constructions in order to ensure proper interactions between FIDO2 flows themselves and also the rest of the Trezor UI.

In particular, this part is suspect: https://github.com/trezor/trezor-firmware/blob/master/core/src/apps/webauthn/fido2.py#L961-L982

Let's investigate options to make this code more readable and thus more reliable. First step seems to make use of `loop.race` to send cancellation/timeout signals, which can then be handled by specific code, as opposed to a generic `finally` clause

**Comment by matejcik** [comment]
The original issue might resolve itself if we implement a nicer event loop for the FIDO process.

There are several sub-problems to this, that are currently solved in a somewhat fragile manner:

1. the FIDO `handle_reports` loop, which is similar to Trezor's `handle_session` handler, must always respond to incoming requests
2. while a FIDO (not U2F) command is in progress, a keep-alive process must run often enough, parallel to waiting for user interaction etc.. This keep-alive process should dynamically respond to what is happening in the original command, so that it provides accurate information: waiting for user input; waiting for calculation; busy with something else; whatever
3. Two FIDO workflows should not run at the same time, there must be some sort of canceling interaction, etc., and this must all interact nicely with the Trezor layout management.

To this mix we are adding soft-lock which is yet another waiting interaction, but this time it must be resolved even before we start processing the FIDO command (otherwise we don't even know if the device is initialized).

There are big similarities to Trezor's normal event loop, so there is a good chance that we can employ same or similar techniques. As an alternative, the current Bitcoin signing generator-based sub-event-loop might be helpful.

cc @andrewkozlik, please correct me if i'm misrepresenting something

**Comment by matejcik** [comment]
while nice to have, there does not seem to be a strong reason to go poking this code. a much better direction would be rewriting FIDO completely in Rust.
